### DISCUSSION TITLE
not getting the unavailable device name

### DISCUSSION THREAD

**Discussion by avigrinberg** [jazzyisj/unavailable-entities-sensor#40] [General]
Hi,

I've installed the script and also the automation and when the automation is running i get a "dot" in the notification.
why?
![image](https://github.com/jazzyisj/unavailable-entities-sensor/assets/74825241/372e2362-b5d3-4faa-a5fc-511f71e99973)

**Comment by jazzyisj** [comment]
Does the reported entity have a friendly name?

<img width="776" alt="image" src="https://github.com/jazzyisj/unavailable-entities-sensor/assets/44064559/87863990-8efd-4b41-8e68-c03f80469878">

**Reply by avigrinberg** [reply]
yes it is.
but now i'm working on the automation alert and i use this code from the automation.
the problem is that the code send me also the ignored entities that i add in the ignored group. where in the code i tell the script to not include the ignored group?

```
`alias: test unavailable entities
description: Create persistent notification if unavailable entities, dismiss if none.
trigger:
  - platform: state
    entity_id: sensor.unavailable_entities
    attribute: entity_id
    to: null
condition:
  - condition: template
    value_template: |
      {{ is_number(trigger.from_state.state)
          and is_number(trigger.to_state.state) }}
action:
  - if:
      - condition: numeric_state
        entity_id: sensor.unavailable_entities
        below: 1
    then:
      - service: persistent_notification.dismiss
        data:
          notification_id: unavailable_entities
    else:
             - service: persistent_notification.create
               data:
                 notification_id: unavailable_entities
                 title: "Unavailable Entities"
                 message: >
                   {% set ns = namespace(result=[]) %}
                   {% for s in expand(state_attr('sensor.unavailable_entities', 'entity_id')) %}
                     {% set ns.result = ns.result + [
                         device_attr(s.entity_id, "name") ~ "|" ~ device_id(s.entity_id) ~ "|- **" ~ s.name ~ "**\n"
                         ~ "  - *entity_id*: " ~ s.entity_id ~ "\n"
                         ~ "  - *state*: " ~ s.state ~ "\n"
                       ]
                     %}
                   {% endfor %}
                   {% set ns.result = ns.result | sort %}
                   {% set lastdev = namespace( id="" ) %}
                   {% set tarr = ns.result %}
                   {% set ns.result = [] %}
                   {% for item in tarr %}
                     {% set dev = namespace( id="" ) %}
                     {% set entity = namespace( data="" ) %}
                     {% set dev.id = item.split("|")[1] %}
                     {% set entity.data = item.split("|")[2] %}
                     {% if lastdev.id != dev.id %}
                       {% if dev.id != 'None' %}
                         {% set ns.result = ns.result + [ "**<a href=\"/config/devices/device/" ~ dev.id ~ "\">" ~ device_attr(dev.id, "name") ~ "</a>**" ] %}
                       {% else %}
                         {% set ns.result = ns.result + [ "**Non-Device Entities**" ] %}
                       {% endif %}
                       {% set lastdev.id = dev.id %}
                     {% endif %}
                     {% set ns.result = ns.result + [ entity.data ] %}
                   {% endfor %}
                   {{ ns.result | join('\n') }}

mode: restart
```
`

**Reply by jazzyisj** [reply]
EDIT:  I forgot that I had the power to do this myself.   Make a note for the future though! :-)

Can you edit this message with code blocks please.  When you add code examples on here and the forums you need to format the code properly so it is understandable and can be copied properly.

<img width="863" alt="image" src="https://github.com/jazzyisj/unavailable-entities-sensor/assets/44064559/dd959d3f-8ec1-4800-a2f5-7a82d81a8fcb">

**Comment by jazzyisj** [comment]
OK so you're asking a completely different question now.  Have you resolved the first question yourself?  You didn't say.

Second question.  You are posting the wrong code.  The issue has to be in your template or group.  It also looks like you imported the automation into the UI editor and the spacing is messed up.  I don't know if it's a cut and paste error or not.

Drag and drop your package file from the file system into the message box here in your reply if you can.  It will upload the entire file so I can look at it.  If you can't do that, copy and paste your entire template sensor and group definition here and format it as code.

**Comment by jazzyisj** [comment]
Closed.  No response from @avigrinberg .   Re-open issue if you still need help.
